### ISSUE TITLE
how can we pass the value at runtime using external functions ?

### ISSUE THREAD

**Issue by gbhangale416** [taverntesting/tavern#209]
###############until.py###############
```python
 def read_mail():
    return verificationCode = 1323213298
```

#####################################

```yaml
 - name: Step_01_07_VerifySetupCodefromMail
    request:
      url: "{host:s}/{service.endPoint:s}/{uid:s}/validate"
      json:
        userName: "{tenatntInfo.setupAdmin:s}"
        verificationCode: 
         $ext:
           function: utils:read_mail
      method: POST
      headers:
        accept: application/json
        Cache-Control: no-cache
    response:
      status_code: 200
```

**Comment by KPilnacek** [comment]
Your `read_mail` function should return a dict. The value will be than accessible in the test under its key name (for more details see [docs](https://taverntesting.github.io/documentation#saving-data-with-external-functions)).

**Comment by gbhangale416** [comment]
> Your `read_mail` function should return a dict. The value will be than accessible in the test under its key name (for more details see [docs](https://taverntesting.github.io/documentation#saving-data-with-external-functions)).
example above is for saving the data but i want to pass the value at runtime

**Comment by michaelboulton** [comment]
The documentation is maybe a bit vague, but you can only use an external function to create the _whole_ body or response - if you just want one value, you should use a pytest fixture (an example is in the [documentation](https://taverntesting.github.io/documentation#usefixtures))

Note that there is also no way to use a fixture to generate a block of json - either use `$ext` to generate the whole request, or a fixture to generate one value. Anything in between is currently not implemented (see #191 )

**Comment by gbhangale416** [comment]
> The documentation is maybe a bit vague, but you can only use an external function to create the _whole_ body or response - if you just want one value, you should use a pytest fixture (an example is in the [documentation](https://taverntesting.github.io/documentation#usefixtures))
> 
> Note that there is also no way to use a fixture to generate a block of json - either use `$ext` to generate the whole request, or a fixture to generate one value. Anything in between is currently not implemented (see #191 )


Value i am going to pass will change at runtime after few steps are executed of my testcase which i will received in mail then read the mail and pass pass the value at runtime so usefixtures i tried using usefixtures but it didn't work bcz fixtures the valued  at the start of testcase
